cmd/prompt/prompt_test.go

```diff
@@ -0,0 +1,241 @@
+					systemMsgFound = true
+				}
+			}
+			require.True(t, userMsgFound, "User message with correct content not found")
+			require.True(t, systemMsgFound, "System message with correct content not found")
+
+			return chatResp, nil
+		}
+
+		// Create a temp prompt file
+		promptContent := `---
+name: Summarizer
+description: Summarizes a given text
+model: gpt-4o-mini
+model_parameters:
+  temperature: 0.5
+---
+
+system:
+You are a text summarizer. Your only job is to summarize a given text to you. I want you to summarize in bullet points.
+
+user:
+Summarize the given text:
+
+<text>
+{{text}}
+</text>
+`
+		tmpDir := t.TempDir()
+		promptFilePath := filepath.Join(tmpDir, "test-prompt.md")
+		err := os.WriteFile(promptFilePath, []byte(promptContent), 0644)
+		require.NoError(t, err)
+
+		buf := new(bytes.Buffer)
+		cfg := command.NewConfig(buf, buf, client, true, 80)
+		promptCmd := NewPromptCommand(cfg)
+		promptCmd.SetArgs([]string{promptFilePath, "text=Test content to summarize"})
+
+		_, err = promptCmd.ExecuteC()
+
+		require.NoError(t, err)
+		require.Equal(t, 1, getChatCompletionCallCount)
+		output := buf.String()
+		require.Contains(t, output, fakeMessageFromModel)
+	})
+
+	t.Run("NewPromptCommand error on missing prompt file", func(t *testing.T) {
+		client := azuremodels.NewMockClient()
+		buf := new(bytes.Buffer)
+		cfg := command.NewConfig(buf, buf, client, true, 80)
+		promptCmd := NewPromptCommand(cfg)
+
+		// Don't provide any args
+		promptCmd.SetArgs([]string{})
+
+		_, err := promptCmd.ExecuteC()
+		require.Error(t, err)
+		require.Contains(t, err.Error(), "prompt file path is required")
+	})
+
+	t.Run("NewPromptCommand prompts for model when not in prompt file", func(t *testing.T) {
+		client := azuremodels.NewMockClient()
+		modelSummary := &azuremodels.ModelSummary{
+			ID:           "test-id-1",
+			Name:         "gpt-4o-mini",
+			FriendlyName: "OpenAI GPT-4o mini",
+			Task:         "chat-completion",
+			Publisher:    "OpenAI",
+			Summary:      "This is a test model",
+			Version:      "1.0",
+			RegistryName: "azure-openai",
+		}
+		client.MockListModels = func(ctx context.Context) ([]*azuremodels.ModelSummary, error) {
+			return []*azuremodels.ModelSummary{modelSummary}, nil
+		}
+
+		// This test can't easily simulate interactive selection, so we'll just check that it
+		// responds with an error when we don't provide a model and it tries to prompt for one
+
+		// Create a temp prompt file without a model
+		promptContent := `---
+name: Summarizer
+description: Summarizes a given text
+---
+
+system:
+You are a text summarizer. Your only job is to summarize a given text to you.
+
+user:
+Summarize the given text:
+
+<text>
+{{text}}
+</text>
+`
+		tmpDir := t.TempDir()
+		promptFilePath := filepath.Join(tmpDir, "test-prompt-no-model.md")
```

Comment:
Should this be a `.prompt.md` file? Additionally should we be checking for that?

```suggestion
		promptFilePath := filepath.Join(tmpDir, "test-prompt-no-model.prompt.md")
```